cli/admin/src/main/java/io/pravega/cli/admin/controller/ControllerCommand.java

```diff
@@ -50,7 +62,38 @@ protected Context createContext() {
         ClientConfig clientConfig = new ClientConfig();
         clientConfig.register(JacksonJsonProvider.class);
         clientConfig.property("sun.net.http.allowRestrictedHeaders", "true");
-        Client client = ClientBuilder.newClient(clientConfig);
+
+        Client client;
+
+        // If tls parameters are configured, set them in client
+        if (getCLIControllerConfig().isTlsEnabled()) {
+            KeyStore ks = null;
+            try {
+                @Cleanup
+                InputStream trustStore = new FileInputStream(new File(getCLIControllerConfig().getTruststore()));
+                ks = KeyStore.getInstance("JKS");
+                ks.load(trustStore, null);
+
+            } catch (KeyStoreException e) {
+                output("The keystore file is invalid, the keystore type is not supported: %s", e.toString());
+            } catch (IOException e) {
+                output("The keystore file is invalid, check if the file exists: %s", e.toString());
+            } catch (NoSuchAlgorithmException e) {
+                output("The keystore file is invalid, the keystore file might be in the wrong format: %s", e.toString());
+            } catch (CertificateException e) {
+                output("The keystore file is invalid, check if the certificates are valid: %s", e.toString());
+            }
+
+            HostnameVerifier controllerHostnameVerifier = new ControllerHostnameVerifier();
+            client = ClientBuilder.newBuilder()
+                    .withConfig(clientConfig)
+                    .trustStore(ks)
+                    .hostnameVerifier(controllerHostnameVerifier)
+                    .build();
+        } else {
+            client = ClientBuilder.newClient(clientConfig);
+        }
+
         // If authorization parameters are configured, set them in the client.
         if (getCLIControllerConfig().isAuthEnabled()) {
```

Comment:
You seem to be using `getCLIControllerConfig()` multiple times. (The name of that method is misleading - it actually does a bit of processing as opposed to returning a cached value that one'd usually expect from a method having a name starting with `get`). You might want to call it once, store the value in a variable and reuse it.

Reply:
Moved the return of `getCLIControllerConfig()` to a single variable which is used everywhere.